src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -23,25 +35,56 @@ public void run()
          for (int tagIndex = 0; tagIndex < OSIsoftConfig.tags.size(); tagIndex++) {
             dataPoints.add(((Tag) OSIsoftConfig.tags.get(tagIndex)).getNewestDataPoints(pointsPerTag));
 
+            String tagName = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagName();
+            if(omfEnabled) {
+               if (tagIndex > 0 ) {
+                  OSIsoftServer.seperateDataMessage();
+               }
+
+               OSIsoftServer.addContainerStartToOMFdataMessage(tagName);
+            }
+
             for(int dataPointIndex = 0; dataPointIndex<((ArrayList) dataPoints.get(tagIndex)).size(); dataPointIndex++)
             {
+
                if(((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex) != null)
                {
-                  OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
+
+                  if(omfEnabled) {
+                     String tagValue = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagValue();

```

Comment:
@oliver-walker-hms  Why are you bybassing the datapoint array and just grabbing the current tag value?

Reply:
still in the loop, so it made more sense to deal with the array out here. For each tag value all i really need to know is the value and timestamp. addPointToBatch does that anyways where it just uses the value in that function and the rest is extra so I just didn't want the extra information

Reply:
fixed this